src/metabase/server/middleware/session.clj

```diff
@@ -305,3 +307,89 @@
   `(do-with-current-user
     (with-current-user-fetch-user-for-id ~current-user-id)
     (fn [] ~@body)))
+
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                              check-session-timeout                                             |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(defsetting session-timeout
+  ;; Should be in the form {:amount 60 :unit "minutes"} where the unit is one of "seconds", "minutes" or "hours". The amount is nillable.
+  (deferred-tru "Time before inactive users are logged out. By default, sessions last indefinitely.")
+  :type       :json
+  :default    nil)
+
+(defn session-timeout->seconds
+  "Convert a session timeout setting to seconds."
+  [{:keys [unit amount]}]
+  (when amount
+    (-> (case unit
+          "seconds" amount
+          "minutes" (* amount 60)
+          "hours"  (* amount 3600))
+        (max 60)))) ; Ensure a minimum of 60 seconds so a user can't lock themselves out
```

Comment:
The use of `(max 60 (* amount 60))` seems pretty off here. Did you mean to use `min` here instead?

Reply:
My idea here was to prevent disaster if the duration in seconds was less than a minute, e.g. 0. It's kind of lying to the user doing it this way though, so maybe it's best remove this and add validation on the FE?

Reply:
oh you are right. i had it backwards